src/main/java/com/arpnetworking/metrics/impl/BaseFileSink.java

```diff
@@ -414,5 +453,7 @@ protected void validate(final List<String> failures) {
         private static final Boolean DEFAULT_PRUDENT = Boolean.FALSE;
         private static final Boolean DEFAULT_DROP_WHEN_QUEUE_FULL = Boolean.FALSE;
         private static final Integer DEFAULT_MAX_QUEUE_SIZE = Integer.valueOf(500);
+        private static final String DEFAULT_MAX_FILE_SIZE = "100MB";
+        private static final String DEFAULT_TOTAL_SIZE_CAP = "20GB";
```

Comment:
Yes, this is a bug in the client configuration. Max history should be the number of files to keep in my opinion. @BrandonArp ?

That means across time and space rolling policies. For example, without a size constraint it should be the number hours. However, with a size constraint it may be fewer than that many hours. 

These configuration options on the client are not intended to duplicate the Logback configuration but to stand on their own.

I suggest:

 - No default maximum size.
 - Setting maximum size sets totalSizeCap on logback to maxHistory times maximum size.
 - Tests for enforcement of max history based on size (since this has been a buggy feature in logback).